third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -4795,8 +4877,14 @@ static void fromJsGetNodeIdByCpdId(const v8::FunctionCallbackInfo<v8::Value>& ar
     Node* node = V8Node::ToImplWithTypeCheck(isolate, nodeObj);
     if (node) {
       // Bind node and get nodeId.
-      auto* domAgent = getOrCreateInspectorDOMAgent(isolate);
-      int nodeId = domAgent->BindDocumentNode(node);
+      auto domAgent = getOrCreateInspectorDOMAgent(isolate);
+      if (!domAgent.has_value()) {
+        recordreplay::CommandDiagnostic("fromJsGetNodeIdByCpdId no context id.");
```

Comment:
There are a few of these new `CommandDiagnostic` calls. Curious:

1. Why only a `CommandDiagnostic` and not a `warning`?
2. Why not `SendMessageToFrontend`?

Note: A `CommandDiagnostic` is only a replay-only `Print` (for now).

Reply:
1) Is this actually a warning?  It's not clear to me, but getting clarity involves watching this over time and understanding if there's something wrong with our implementation, or if all this is just fine.  Also, warnings generate stacks, which take some time to make.
2) SendMessageToFrontend needs to happen in the context of a CDP call; `fromJsGetNodeIdByCpdId` is invoked directly from the JS.  Now, whether this should throw or not--I'm not sure.  So, for now, we log the issue.

Reply:
> watching this over time

That's definitely important!
Are you watching this? Maybe we can discuss this today or in the coming days.